src/porepy/numerics/fv/mpfa.py

```diff
@@ -1379,7 +1141,50 @@ def _create_bound_rhs(
 
         rhs_bound = sps.vstack([neu_rob_cell, dir_cell]) * bnd_2_all_hf
 
-        return rhs_bound
+        return rhs_bound.tocsr()
+
+    def _bc_for_subgrid(
+        self, bc: pp.BoundaryCondition, sub_g: pp.Grid, face_map: np.ndarray
+    ) -> pp.BoundaryCondition:
+        """ Obtain a representation of a boundary condition for a subgrid of
+        the original grid.
+
+        This is somehow better fit for the BoundaryCondition class, but it is not clear
+        whether the implementation is sufficiently general to be put there.
+
+        Parameters:
+            sub_g (pp.Grid): Grid for which the new condition applies. Is
+                assumed to be a subgrid of the grid to initialize this object.
+            face_map (np.ndarray): Index of faces of the original grid from
+                which the new conditions should be picked.
+
+        Returns:
+            BoundaryConditionVectorial: New bc object, aimed at a smaller grid.
+            Will have type of boundary condition, basis and robin_weight copied
+            from the specified faces in the original grid.
+
+        """
+
+        sub_bc = pp.BoundaryCondition(sub_g)
+
+        sub_bc.is_dir = bc.is_dir[face_map]
+        sub_bc.is_rob = bc.is_rob[face_map]
+        sub_bc.is_neu[sub_bc.is_dir + sub_bc.is_rob] = False
+
+        sub_bc.robin_weight = bc.robin_weight[face_map]
+        sub_bc.basis = bc.basis[face_map]
+
+        return sub_bc
+
+    def _constit_for_subgrid(
+        self, constit: pp.SecondOrderTensor, loc_cells: np.ndarray
+    ) -> pp.SecondOrderTensor:
+        # Copy stiffness tensor, and restrict to local cells
+        loc_c = constit.copy()
+        loc_c.values = loc_c.values[::, ::, loc_cells]
+        # Also restrict the lambda and mu fields; we will copy the stiffness
+        # tensors later.
```

Comment:
I don't understand the comment. Does this mean that the function is yet to be expanded to fourth order tensors? TODO?

Reply:
Leftover from copy from the similar funciton in mpsa. My bad.